File: BackEnd/src/main/java/com/example/demo/models/EvenementChantier.java
package com.example.demo.models;


import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
@Entity
@Table(name = "evenement")
public class EvenementChantier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Date start_date;
    private Date end_date;
    private String text;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @ManyToOne
    private Chantier chantier;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @ManyToOne
    private Vehicule vehicule;
    @ManyToMany(cascade = CascadeType.REMOVE)
    private List<Tache> taches;
    @ManyToMany(cascade = CascadeType.REMOVE)
    private List<Ouvrier> ouvriers;


}

File: FrontEnd/src/app/services/models/evenement-chantier.ts
/* tslint:disable */
/* eslint-disable */
import { Ouvrier } from '../models/ouvrier';
import { Tache } from '../models/tache';
export interface EvenementChantier {
  chantier?: number;
  end_date?: string;
  id?: number;
  ouvriers?: Array<Ouvrier>;
  start_date?: string;
  taches?: Array<Tache>;
  text?: string;
  vehicule?: number;
}
